website/source/docs/auth/aws-iam.html.md

```diff
@@ -0,0 +1,606 @@
+  The aws-iam backend allows automated authentication AWS IAM principals.
+---
+
+# Auth Backend: aws-iam
+
+The aws-iam auth backend provides a secure introduction mechanism for AWS IAM
+principals. This allows you to reduce the problem of securely introducing a
+Vault token to the problem of securely introducing AWS IAM credentials, which
+AWS has already solved in a number of use cases, such as via EC2 Instance
+Profiles and IAM roles attached to Lambda functions. It also allows you to have
+a consistent workflow between developers working on a local laptop with tools
+such as Hologram.
+
+## Comparison with AWS-EC2 Authentication Backend
+
+The AWS-IAM and AWS-EC2 authentication backends serve similar purposes. Both
+look to authenticate some type of AWS entity to Vault. However, in many ways,
+the similarity ends there. The following is a comparison of the two entities:
+
+* What type of entity is authenticated:
+  * The AWS-EC2 backend authenticates AWS EC2 instances only.
+  * The AWS-IAM backend authenticates AWS IAM principals. This can include
+    IAM users, IAM roles assumed from other accounts, AWS Lambdas that are
+    launched in an IAM role, or even EC2 instances that are launched in an
+    EC2 instance profile.
+* How the entities are authenticated
+  * The AWS-EC2 backend authenticates instances by making use of the EC2 instance
+    identity document, which is a cryptographically signed document containing
+    metadata about the instance. This document changes relatively infrequently,
+    so Vault adds a number of other constructs to mitigate against replay
+    attacks, such as client nonces, role tags, instance migrations, etc.
+  * The AWS-IAM backend authenticates by having clients provide a specially
+    signed AWS API request which the backend then passes on to AWS to validate
+    the signature and tell Vault who created it. The actual secret (i.e.,
+    the AWS secret access key) is never transmitted over the wire, and the
+    AWS signature algorithm automatically expires requests after 15 minutes,
+    providing simple and robust protection against replay attacks.
+* Specific use cases
+  * If you have a long-lived EC2 instance which you are unable to relaunch into
+    an IAM instance profile, then the AWS-EC2 backend is probably the best
+    solution for you. (While you could store long-lived AWS IAM user
+    credentials on disk and use those to authenticate to Vault, that would not
+    be recommended.)
+  * If you have non-EC2 instance entities, such as IAM users, Lambdas in IAM
+    roles, or developer laptops using [AdRoll's Hologram](https://github.com/AdRoll/hologram)
+    then you would need to use the AWS-IAM backend.
+  * If you have EC2 instances which are already in an IAM instance profile, then
+    you could use either backend.
+
+## Authentication Workflow
+
+AWS has introduced the [`sts:GetCallerIdentity`](http://docs.aws.amazon.com/STS/latest/APIReference/API_GetCallerIdentity.html)
```

Comment:
This reads weirdly. Of course AWS has introduced it, because that's what this backend is based on...